src/nvs.rs

```diff
@@ -326,149 +411,68 @@ impl<T: NvsPartitionId> EspNvs<T> {
         Ok(Self(partition, handle))
     }
 
+    #[cfg(all(
```

Comment:
Instead of all of `contains`, `contains_key_type`, `find_key_type`, why not just one - `find_key` - which does what `find_key_type` currently does? Less is more?
- Using `find_key` for "contains such  key regardless of the type" would be as simple as `find_key()?.is_some()`
- Using `find_key` for "contains such key with type XYZ" would be `matches!(find_key()?, Some(NvsDataType::U8))`

... so I don't think we need the extra syntax sugar.